File: FrontEnd-Angular/src/app/services/despesa/despesa.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080";

@Injectable({
  providedIn: 'root'
})
export class DespesaService {
  constructor(private http: HttpClient) { }

  postDespesa(despesaDTO: any): Observable<any> {
    return this.http.post(`${BASIC_URL}/api/despesa`, despesaDTO);
  }

  getAllDespesas(): Observable<any> {
    return this.http.get(`${BASIC_URL}/api/despesa/all`);
  }

  deleteDespesa(id: number): Observable<any> {
    return this.http.delete(`${BASIC_URL}/api/despesa/${id}`);
  }

  getDespesaById(id: number): Observable<any> {
    return this.http.get(`${BASIC_URL}/api/despesa/${id}`);
  }

  atualizacaoDespesa(id: number, despesaDTO: any): Observable<any> {
    return this.http.put(`${BASIC_URL}/api/despesa/${id}`, despesaDTO);
  }
}

File: Backend/ProjetoFinal/src/main/java/com/Financas/ProjetoFinal/services/Despesa/DespesaService.java
package com.Financas.ProjetoFinal.services.Despesa;

import com.Financas.ProjetoFinal.dto.DespesaDTO;
import com.Financas.ProjetoFinal.entity.Despesa;

import java.util.List;

public interface DespesaService {

    Despesa postDespesa(DespesaDTO despesaDTO);


    public List<Despesa> getAllDespesas();


    Despesa getDespesaById(Long id);


    Despesa updateDespesa(Long id, DespesaDTO despesaDTO);


    void deleteDespesa(Long id);

}
